packages/contracts-bedrock/STYLE_GUIDE.md

```diff
@@ -106,6 +132,13 @@ making changes to a well-known, simple, and recognizable contract.
 Where basic functionality is already supported by an existing contract in the OpenZeppelin library,
 we should default to using the Upgradeable version of that contract.
 
+### Source Code
+
+The following guidelines should be followed for all contracts in the `src/` directory:
+
+- All state change should have a corresponding event emitted. This ensures that all state changes are
+  transparent, can be easily monitored, and can be reconstructed from the event logs.
```

Comment:
This might be overly broad. 
ie. some methods, particularly in the bridge path, will make many state changes, but it's sufficient to emit a single event. 

Could it make sense to say 
```suggestion
- All state changing functions should emit a corresponding event. This ensures that all actions are transparent, can be easily monitored, and can be reconstructed from the event logs.
```

Reply:
If "reconstructed from the event logs" is important, we may need to provide more guidance about the specific arguments/data the event should include.

Reply:
> If "reconstructed from the event logs" is important

I don't think this is super important, but it is a good heuristic to help decide what to emit. Would you update your suggestion given that? (I'm also ok with your suggestion as-is)